chanzuckerberg/idseq-web
# Charles/pipeline separation and input chunking

# Description
- Breaks the pipeline into 3 separate stages (3 scripts: host_filtering.py, non_host_alignment.py, postprocess.py; all 3 use functions from common.py).
- For gsnap/rapsearch steps, breaks input files into small chunks and processes chunks sequentially -- each chunk will have to wait for resources independently once the previous chunk is finished. That way smaller jobs can get into gsnap/rapsearch processing even before an earlier longer job is completely done.
- Pipeline kickoff commands in the rails app will need to be changed to use these new scripts. The folder structure in S3 will need to be changed: the "results" folder needs to be split into a "stage 1 results" and a "stage 2 results" folder for lazy-run to work efficiently (avoid copying large files from stage 1 when executing stage 2).

## Type of change
- New feature (non-breaking change which adds functionality)
- Breaking change (fix a feature that knowingly causes existing functionality to not work as expected)

# Test and Reproduce
For each one of the three scripts :
1. Inspect the main() function of the script for the environment variables it uses.
2. Run the script with these environment variables set as needed for an example sample of your choice. E.g.:
```
BASE_PATH=s3://czbiohub-idseq-samples-alpha/samples/11/199;
sudo INPUT_BUCKET=$BASE_PATH/results OUTPUT_BUCKET=$BASE_PATH/postprocess python postprocess.py
```
3. Check if script runs without errors and produces the desired outputs.

## Impacted Areas in Application
Data processing between upload and report display.

# To be tested extensively
